### DISCUSSION TITLE
How to use env variable in Gatsby client side?

### DISCUSSION THREAD

**Discussion by orvillelim** [vercel/vercel#4651] [Help] [answered]
What I'm trying to do is set up dev/staging with Vercel, so I need to use staging keys for my Gatsby app (client-side).

**What I have tried:**

 - Add environment Preview variable from Vercel project settings and connect Gitlab for integration. Then push test commits in dev to test if env is working, but it did not.
This should work base on this thread: https://github.com/vercel/vercel/issues/2613#issuecomment-594036619

- Use Vercel CLI to deploy, it was able to expose the environment variable but it uses the production variables. I believe it's because from Build step npm run build. but then how should I expose staging env?

https://stackoverflow.com/questions/62371066/how-to-use-vercel-env-variable-in-gatsby-js-client-side

**Comment by styfle** [comment, marked as answer]
Gatsby requires a prefix of `GATSBY_` in order to bundle environment variables.

https://www.gatsbyjs.org/docs/environment-variables/#example-of-using-an-environment-variable

Related to #4405 and #3979.

> Use Vercel CLI to deploy, it was able to expose the environment variable but it uses the production variables.

Do you have a `.env` file? Try creating a `.vercelignore` file to ignore the `.env` file.

**Reply by orvillelim** [reply]
Yeah, I was able to figure it out after many hours lol. Thanks @styfle. Closing this now
